examples/light-switch-app/silabs/src/LightSwitchMgr.cpp

```diff
@@ -26,18 +26,123 @@
 
 #include "AppConfig.h"
 #include "AppEvent.h"
+#include "AppTask.h"
 
 #include <app-common/zap-generated/attributes/Accessors.h>
 #include <app-common/zap-generated/cluster-objects.h>
 #include <app/clusters/switch-server/switch-server.h>
 #include <platform/CHIPDeviceLayer.h>
+#include <cmsis_os2.h>
+#include <platform/silabs/platformAbstraction/SilabsPlatform.h>
 
 using namespace chip;
 using namespace chip::app;
 using namespace chip::app::Clusters;
+using namespace chip::DeviceLayer;
+using namespace chip::DeviceLayer::Silabs;
 
 LightSwitchMgr LightSwitchMgr::sSwitch;
 
+AppEvent CreateNewEvent(AppEvent::AppEventTypes type)
+{
+    AppEvent aEvent;
+    aEvent.Type                      = type;
+    aEvent.Handler                   = LightSwitchMgr::GeneralEventHandler;
+    LightSwitchMgr * lightSwitch         = &LightSwitchMgr::GetInstance();
+    aEvent.LightSwitchEvent.Context  = lightSwitch;
+    return aEvent;
+}
+
+void LightSwitchMgr::Timer::Start()
+{
+    // Starts or restarts the function timer
+    if (osTimerStart(mHandler, pdMS_TO_TICKS(100)) != osOK)
```

Comment:
This logic seems odd to force 100 as the timeout value. Can you provide mode information on the logic?

Reply:
I have picked up the timer logic from window-app.